pkg/kubelet/kubelet_test.go

```diff
@@ -735,21 +733,12 @@ func TestHandlePodRemovesWhenSourcesAreReady(t *testing.T) {
 	}
 }
 
-type testNodeLister struct {
-	nodes []*v1.Node
-}
-
-func (nl testNodeLister) Get(name string) (*v1.Node, error) {
-	for _, node := range nl.nodes {
-		if node.Name == name {
-			return node, nil
-		}
+func testNodeLister(nodes ...*v1.Node) corelisters.NodeLister {
+	store := cache.NewIndexer(cache.MetaNamespaceKeyFunc, cache.Indexers{})
+	for _, n := range nodes {
+		store.Add(n)
```

Comment:
this is the reason of the churn on the test fails, but IMHO this is "more realistic"